File: fe-arise/src/dtos/invoice/InvoiceDTO.ts
import { AuditInfoDTO } from "../base";
import { DiscountDTO } from "../discount";

export type InvoiceDTO = {
  id: number;
  studentNickName: string;
  discount: DiscountDTO;
  studentPhoneNumber: string;
  invoiceAttempt: number;
  staffName: string;
  studentName: string;
  studentCode: string;
  classCode: string;
  className: string;
  tuitionOwed: number;
  tuitionCourseOwed: number;
  invoiceDiscount: number;
  amount: number;
  paymentType: PaymentTypeConstants;
  description: string;
  invoiceStatus: InvoiceStatusConstants;
  startDate: Date;
} & AuditInfoDTO;

export enum InvoiceStatusConstants {
  NOT_PAID = "NOT_PAID",
  PARTIALLY_PAID = "PARTIALLY_PAID",
  FULLY_PAID = "FULLY_PAID",
}
export function formatInvoiceStatus(status?: InvoiceStatusConstants): string {
  if (!status) return "";

  return status
    .toLowerCase()
    .split("_")
    .map((word) => word.charAt(0).toUpperCase() + word.slice(1))
    .join(" ");
}

export enum PaymentTypeConstants {
  CASH = "CASH",
  BANK_TRANSFER = "BANK_TRANSFER",
}

File: be-arise/src/main/java/vn/codezx/arise/dtos/invoice/InVoiceDTO.java
package vn.codezx.arise.dtos.invoice;

import java.util.Date;
import lombok.AllArgsConstructor;
import lombok.EqualsAndHashCode;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;
import lombok.experimental.SuperBuilder;
import vn.codezx.arise.constants.InvoiceStatus;
import vn.codezx.arise.constants.PaymentTypeConstants;
import vn.codezx.arise.entities.base.BaseInfo;
import vn.codezx.arise.entities.setting.Discount;

@AllArgsConstructor
@NoArgsConstructor
@Getter
@Setter
@SuperBuilder
@EqualsAndHashCode(callSuper = false)
public class InVoiceDTO extends BaseInfo {
  private int id;
  private String studentNickName;
  private String studentPhoneNumber;
  private Discount discount;
  private String staffName;
  private String studentName;
  private String studentCode;
  private String classCode;
  private String className;
  private int invoiceAttempt;
  private float tuitionOwed;
  private float tuitionCourseOwed;
  private float amount;
  private float invoiceDiscount;
  private PaymentTypeConstants paymentType;
  private String description;
  private InvoiceStatus invoiceStatus;
  private Date startDate;
}
